arches_component_lab/views/api/card_x_node_x_widget.py

```diff
@@ -125,4 +133,12 @@ def get(self, request, graph_slug, nodegroup_alias):
             serialize_card_x_node_x_widget(widget_instance, datatype_factory)
             for widget_instance in widget_instances
         ]
+
+        # Update component paths to declared arches_component_lab widget paths
+        widget_mappings = WidgetMapping.objects.all()
+        for serialized_datum in serialized_data:
+            serialized_datum["widget"]["component"] = widget_mappings.get(
+                widget_id=serialized_datum["widget"]["widgetid"]
+            ).component
```

Comment:
Instead of queries in a loop, I'd just get a dict first and check that in the loop, e.g. `WidgetMapping.objects.in_bulk()`